security/pkg/k8s/chiron/utils.go

```diff
@@ -0,0 +1,327 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chiron
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"reflect"
+	"time"
+
+	"k8s.io/api/admissionregistration/v1beta1"
+
+	"github.com/ghodss/yaml"
+
+	"istio.io/istio/pkg/spiffe"
+
+	"istio.io/istio/security/pkg/pki/util"
+	"istio.io/pkg/log"
+
+	cert "k8s.io/api/certificates/v1beta1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	admissionreg "k8s.io/client-go/kubernetes/typed/admissionregistration/v1beta1"
+)
+
+// Generate a certificate and key from k8s CA
+// Working flow:
+// 1. Generate a CSR
+// 2. Submit a CSR
+// 3. Approve a CSR
+// 4. Read the signed certificate
+// 5. Clean up the artifacts (e.g., delete CSR)
+func genKeyCertK8sCA(wc *WebhookController, secretName string, secretNamespace, svcName string) ([]byte, []byte, error) {
```

Comment:
Put this into pkg/k8s?

Reply:
This function has been placed into security/pkg/k8s/chiron/utils.go.